packages/fixie-sdk/src/types.ts

```diff
@@ -0,0 +1,48 @@
+export interface MessageBase {
+  metadata?: Record<string, string | number | boolean | object | null>;
+}
+
+export interface FunctionCallMessage extends MessageBase {
+  kind: 'functionCall';
+  name: string;
+  args: Record<string, string | number | boolean | null>;
+}
+
+export interface FunctionResponseMessage extends MessageBase {
+  kind: 'functionResponse';
+  name: string;
+  response: string;
+  failed: boolean;
+}
+
+export interface TextMessage extends MessageBase {
+  kind: 'text';
+  content: string;
+}
+
+export type Message = FunctionCallMessage | FunctionResponseMessage | TextMessage;
+
+export interface ConversationTurn {
+  id: string;
+  timestamp: string;
+  role: 'user' | 'assistant';
+  messages: Message[];
+  state: 'in-progress' | 'done' | 'stopped' | 'error';
+  metadata?: Record<string, string | number | boolean | object | null> | null;
```

Comment:
is this where things like tz offset will get stuffed?

Reply:
This was covered by our in-person convo, right?

Reply:
Yes, though I didn't add the time zone offset context thing.

I did wire up the OpenAI model selection and proxy though.